includes/service-providers/mailchimp/class-newspack-newsletters-mailchimp.php

```diff
@@ -1793,24 +1806,7 @@ public function add_contact( $contact, $list_id = false, $tags = [], $interests
 
 			// Create or update a list member.
 			$member_hash = Mailchimp::subscriberHash( $email_address );
-			$result = $mc->put( "lists/$list_id/members/$member_hash", $update_payload );
-
-			if (
-				! $result ||
-				! isset( $result['status'] ) ||
-				// See Mailchimp error code glossary: https://mailchimp.com/developer/marketing/docs/errors/#error-glossary.
-				(int) $result['status'] >= 400 ||
-				( isset( $result['errors'] ) && count( $result['errors'] ) )
-			) {
-				return new WP_Error(
-					'newspack_newsletters_mailchimp_add_contact_failed',
-					sprintf(
-						/* translators: %s: Mailchimp error message */
-						__( 'Failed to add contact to list. %s', 'newspack-newsletters' ),
-						isset( $result['detail'] ) ? $result['detail'] : ''
-					)
-				);
-			}
+			$result      = $this->validate( $mc->put( "lists/$list_id/members/$member_hash", $update_payload ) );
```

Comment:
@leogermani I think this change is still safe because it should result in the same behavior and error code—the removed code was mainly duplicating some of the checks in the `validate()` method.